File: valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        # Eğer iki dize farklı uzunluktaysa, anagram olamazlar.
        if len(s) != len(t):
            return False
        
        # Karakter sayılarını tutacak sözlükler oluşturuluyor.
        s_count = {}  # İlk dize için karakter sayıları
        t_count = {}  # İkinci dize için karakter sayıları

        # İlk dizeyi tarayarak karakter sayılarını güncelle
        for char in s:
            # get() metodu, belirtilen anahtarın değerini döndürür.
            # Eğer anahtar yoksa, varsayılan olarak 0 değerini döndürür.
            s_count[char] = s_count.get(char, 0) + 1

        # İkinci dizeyi tarayarak karakter sayılarını güncelle
        for char in t:
            t_count[char] = t_count.get(char, 0) + 1

        # İki dizenin karakter sayılarını karşılaştır.
        return s_count == t_count

File: valid-anagram.js
var isAnagram = function(s, t) {
  // Eğer iki dize farklı uzunluktaysa, anagram olamazlar.
  if (s.length !== t.length) return false;

  // 26 elemanlı bir diziyi sıfırlarla doldur.
  const charMap = Array(26).fill(0);

  // Her iki diziyi tarayarak karakter sayılarını güncelle.
  for (let i = 0; i < s.length; i++) {
      // Karakterlerin ASCII değerlerini kullanarak 0'dan başlayan indekslere ekleme veya çıkarma yap.
      charMap[s.charCodeAt(i) - 'a'.charCodeAt(0)]++;
      charMap[t.charCodeAt(i) - 'a'.charCodeAt(0)]--;
  }

  // charMap dizisindeki her elemanın 0 olup olmadığını kontrol et.
  // Eğer her eleman 0 ise, iki dize anagramdır.
  return charMap.every(count => count === 0);
};
